### PR TITLE
Vtcca 2885

### PR DESCRIPTION
**Pull request by jon-pike**

### CODE REVIEW

File: app/connectors/propertyLinking/PropertyLinkConnector.scala [hunk 1]
```diff
@@ -95,15 +95,12 @@ class PropertyLinkConnector @Inject()(config: ServicesConfig, http: WSHttp)(impl
...
-
     // filter agents on representationStatus
-    ownerAuthResult.map(oar =>
-      oar.copy(authorisations = oar.authorisations.map(auth =>
-        auth.copy(agents = auth.agents.map(ags => ags.filter(ag => validAgent(ag)))))))
+    filterAgents(ownerAuthResult, representationStatusFilter)
+  }
+
+  def validAgent(agent: OwnerAuthAgent, representationStatusFilter: Seq[RepresentationStatus]): Boolean = {
+    representationStatusFilter.map(_.name.toUpperCase).contains(agent.status.toUpperCase)
```
Comment (DanielConnelly) [inline app/connectors/propertyLinking/PropertyLinkConnector.scala]: there is an easier way to write this. 
with ignore case.

File: app/connectors/propertyLinking/PropertyLinkConnector.scala [hunk 2]
```diff
@@ -72,38 +72,12 @@ class PropertyLinkConnector @Inject()(config: ServicesConfig, http: WSHttp)(impl
...
-          "startPoint" -> pagination.startPoint.toString,
-          "pageSize" -> pagination.pageSize.toString,
-          "requestTotalRowCount" -> pagination.requestTotalRowCount.toString)
-    )
-
-    def validAgent(agent: OwnerAuthAgent): Boolean =
-      agent.status.fold(false) { status =>
-        representationStatusFilter.map(_.name.toUpperCase).contains(status.toUpperCase)
-      }
+    filterAgents(ownerAuthResult, representationStatusFilter)
```
Comment (DanielConnelly) [inline app/connectors/propertyLinking/PropertyLinkConnector.scala]: dont think this works the way you think it do?

File: app/controllers/agentAppointment/AppointAgentController.scala [hunk 3]
```diff
@@ -36,19 +32,17 @@ import play.api.Logger
...
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class AppointAgentController @Inject()(
                                         representations: PropertyRepresentationConnector,
                                         accounts: GroupAccounts,
-                                        propertyLinks: PropertyLinkConnector,
                                         agentsConnector: AgentsConnector,
                                         authenticated: AuthenticatedAction,
                                         appointRevokeService: AppointRevokeAgentService,
```
Comment (DanielConnelly) [inline app/controllers/agentAppointment/AppointAgentController.scala]: The service should ideally be named after the business case it is filling, Aka Agent Relationship, I missed in last PR.

File: app/connectors/propertyLinking/PropertyLinkConnector.scala (hunk 2 above)
Comment (jon-pike) [reply app/connectors/propertyLinking/PropertyLinkConnector.scala]: as discussed

File: app/connectors/propertyLinking/PropertyLinkConnector.scala (hunk 1 above)
Comment (jon-pike) [reply app/connectors/propertyLinking/PropertyLinkConnector.scala]: done

File: app/controllers/agentAppointment/AppointAgentController.scala (hunk 3 above)
Comment (jon-pike) [reply app/controllers/agentAppointment/AppointAgentController.scala]: done

File: app/controllers/agentAppointment/AppointAgentController.scala [hunk 4]
```diff
@@ -216,15 +204,16 @@ class AppointAgentController @Inject()(
       accounts.withAgentCode(agentCode.toString) flatMap {
         case Some(group) =>
           for {
-            response <- propertyLinks.getMyOrganisationsPropertyLinks(GetPropertyLinksParameters(
+            response <- agentRelationshipService.getMyOrganisationsPropertyLinks(GetPropertyLinksParameters(
```
Comment (DanielConnelly) [inline app/controllers/agentAppointment/AppointAgentController.scala]: should this be using the get my organisation property links.

File: app/services/AgentRelationshipService.scala [hunk 5]
```diff
@@ -115,6 +133,31 @@ class AppointRevokeAgentService @Inject()(representations: PropertyRepresentatio
...
+                logger.warn(s"Agent $agentCode for the property link with subission ID $pLink doesn't exist in cache - this shouldn't be possible.")
+                Future.failed(new AppointRevokeException((s"Agent $agentCode for the property link with subission ID $pLink doesn't exist in cache - this shouldn't be possible.")))
+            }
+          case None =>
+            logger.warn(s"Session ID $sessionId no longer in property links cache - should be redirected to login by auth.")
+            Future.failed(new AppointRevokeException(s"Session ID $sessionId no longer in property links cache - should be redirected to login by auth."))
+        }
+      case None =>
+        logger.warn(s"Unable to obtain session ID from request to retrieve property links cache - should be redirected to login by auth.")
+        Future.failed(new AppointRevokeException(s"Unable to obtain session ID from request to retrieve property links cache - should be redirected to login by auth."))
```
Comment (DanielConnelly) [inline app/services/AgentRelationshipService.scala]: we dont appear to catch these errors?

File: test/services/AgentRelationshipServiceSpec.scala [hunk 6]
```diff
@@ -108,7 +140,20 @@ class AppointRevokeAgentServiceSpec extends ServiceSpec {
...
-  "createAndSubmitAgentRepRequest" should "return option unit when session id can't be obtained (handled by auth)" in {
+  "createAndSubitAgentRevokeRequest" should "throw exception when session id doesn't exist in cache (handled by auth)" in {
+
+    when(mockSessionRepo.get[SessionPropertyLinks](any(), any())).thenReturn(Future.successful(None))
+
+    val res = testService.createAndSubitAgentRevokeRequest(
+      List("999", "8888"),
+      5L)
+
+    ScalaFutures.whenReady(res.failed) { ex =>
```
Comment (DanielConnelly) [inline test/services/AgentRelationshipServiceSpec.scala]: event ScalaFuture and use .futureValue.

File: app/services/AgentRelationshipService.scala (hunk 5 above)
Comment (jon-pike) [reply app/services/AgentRelationshipService.scala]: specific exception now caught

File: test/services/AgentRelationshipServiceSpec.scala (hunk 6 above)
Comment (jon-pike) [reply test/services/AgentRelationshipServiceSpec.scala]: done

**Comment by DanielConnelly** [comment]
LGTM 👍

**Comment by reddy-yattapu** [comment]
:+1:

**Pull request merged**